modules/layers/src/polygon-layer/polygon-layer.ts

```diff
@@ -19,198 +19,56 @@
-   * features if the stroked attribute is true.
-   *
-   * @default 1
-   */
-  lineWidthScale?: number;
-
-  /**
-   * The minimum line width in pixels.
-   *
-   * @default 0
-   */
-  lineWidthMinPixels?: number;
-
-  /**
-   * The maximum line width in pixels
-   *
-   * @default Number.MAX_SAFE_INTEGER
-   */
-  lineWidthMaxPixels?: number;
-
-  /**
-   * Type of joint. If `true`, draw round joints. Otherwise draw miter joints.
-   *
-   * @default false
-   */
-  lineJointRounded?: boolean;
-
-  /**
-   * The maximum extent of a joint in ratio to the stroke width.
-   *
-   * Only works if `lineJointRounded` is false.
-   *
-   * @default 4
-   */
-  lineMiterLimit?: number;
-
-  lineDashJustified?: boolean;
-
-  /** Called on each object in the data stream to retrieve its corresponding polygon. */
-  getPolygon?: AccessorFunction<DataT, any>;
-
-  /**
-   * Fill collor value or accessor.
-   *
-   * @default [0, 0, 0, 255]
-   */
-  getFillColor?: Accessor<DataT, Color>;
-
-  /**
-   * Line color value or accessor.
-   *
-   * @default [0, 0, 0, 255]
-   */
-  getLineColor?: Accessor<DataT, Color>;
-
-  /**
-   * Line width value or accessor.
-   *
-   * @default [0, 0, 0, 255]
-   */
-  getLineWidth?: Accessor<DataT, number>;
-
-  /**
-   * Elevation valur or accessor.
-   *
-   * Only used if `extruded: true`.
-   *
-   * @default 1000
-   */
-  getElevation?: Accessor<DataT, number>;
-
-  /**
-   * This property has been moved to `PathStyleExtension`.
-   *
-   * @deprecated
-   */
-  getLineDashArray?: Accessor<DataT, number> | null;
-
-  /**
-   * If `false`, will skip normalizing the coordinates returned by `getPolygon`.
-   *
-   * **Note**: This prop is experimental
-   *
-   * @default true
-   */
-  _normalize?: boolean;
-
-  /**
-   * Specifies the winding order of rings in the polygon data.
-   *
-   * **Note**: This prop is experimental
-   *
-   * @default 'CW'
-   */
-  _windingOrder?: 'CW' | 'CCW';
-
-  /**
-   * Material props for lighting effect.
-   *
-   * @default true
-   * @see https://deck.gl/docs/developer-guide/using-lighting#constructing-a-material-instance
-   */
-  material?: true | MaterialProps | null;
-};
+export type _PolygonLayerProps<DataT = any> = FillProps<DataT> &
+  StrokeProps<DataT> &
+  Polygon3DProps<DataT> & {
+    /** Called on each object in the data stream to retrieve its corresponding polygon. */
+    getPolygon?: AccessorFunction<DataT, any>;
+
+    /**
+     * If `false`, will skip normalizing the coordinates returned by `getPolygon`.
+     *
+     * **Note**: This prop is experimental
+     *
+     * @default true
+     */
+    _normalize?: boolean;
+
+    /**
+     * Specifies the winding order of rings in the polygon data.
+     *
+     * **Note**: This prop is experimental
+     *
+     * @default 'CW'
+     */
+    _windingOrder?: 'CW' | 'CCW';
+
+    /** @deprecated */
+    lineDashJustified?: boolean;
```

Comment:
Just drop the deprecated props to encourage following the latest API

Reply:
Well, i would discuss. I think that typing something as `@deprecated` is enough signal to discourage use of this feature.

My understanding of deprecation is feature _still_ works, but is planned to be removed or discouraged.

Implementation still supports this prop (brief look, correct me if i'm wrong) so i would rather have it explicitly typed as deprecated that removed.

Reply:
Second thing, is that impl _uses_ this like this:

const {
     ...
      lineDashJustified
    } = this.props;
    
and to remove this prop from `_PolygonLayerProps` i would need to add unsafe workarounds (like `as any` or @ts-expect-error) or some very explit typings.

Reply:
`lineDashJustified` no longer works after we move the dash styling from `PathLayer` to `PathStyleExtension`. I think these are code left over from the refactor.